packages/core/src/common/audioPlayer.ts

```diff
@@ -0,0 +1,112 @@
+import { Howl } from 'howler';
+import { leastSevere, mostSevere, defaultAudioSrc } from './constants';
+import { AudioPlayerInterface, AudioPlayerConfig } from './types';
+
+export const calculateSeverity = (severity: number): number => {
+  return Math.max(Math.min(severity, leastSevere), mostSevere);
+};
+
+export const calculateMaxVolume = (maxVolume: number): number => {
+  return Math.max(Math.min(maxVolume, 1.0), 0.0);
+};
+export class AudioPlayer implements AudioPlayerInterface {
+  config: AudioPlayerConfig;
+  private soundID: number | undefined;
+  private player: Howl | undefined;
+  private severity: number | undefined;
+
+  constructor() {
+    this.config = {
+      isMuted: false,
+      isPlaying: false,
+      maxVolume: 1.0,
+    };
+  }
+
+  public isPlaying(): boolean {
+    return this.config.isPlaying;
+  }
+
+  public isMuted(): boolean {
+    return this.config.isMuted;
+  }
+
+  public mute(): void {
+    if (this.player && this.soundID) {
```

Comment:
what if soundID is 0? Or a negative number?

Reply:
negative numbers are truthy values in typescript, but will add the case that `soundID` is 0